peekingduck/pipeline/nodes/draw/instance_mask.py

```diff
@@ -0,0 +1,389 @@
+            masked_area_colored = cv.addWeighted(
+                coloured_seg_mask, ALPHA, masked_area, 1 - ALPHA, 0
+            )
+
+            # get the inverted mask i.e. image outside of the masked area
+            mask_inv = 1 - masks[index]
+            # remove masked area from image to be returned
+            ret_image = cv.bitwise_and(ret_image, ret_image, mask=mask_inv)
+            ret_image = cv.add(ret_image, masked_area_colored)
+
+            if self.config["show_contours"]:
+                ret_image = self._draw_contours_single_mask(
+                    masks, index, ret_image, color
+                )
+
+        return ret_image
+
+    def _get_instance_color(self, instance_class: str) -> Tuple[int, int, int]:
+        """Returns color to use for next segmentation instance, depending on
+        chosen color scheme:
+
+        "conventional" - Random color.
+
+        "hue_family" - Each object class uses a certain pre-assigned hue. Then
+            each new instance will use the same hue, but with a slightly
+            different saturation.
+
+        Args:
+            instance_class (str): Class of detected object.
+
+        Returns:
+            Tuple[int,int,int]: Color to use for next instance, in RGB.
+        """
+        if self.config["instance_color_scheme"] == "conventional":
+            color_hsv = (randint(0, 179), randint(100, 255), 255)
+            color = self._hsv_to_rgb(color_hsv)
+        elif self.config["instance_color_scheme"] == "hue_family":
+            color = self.class_instance_color_state.get(instance_class)  # type: ignore
+            if not color:
+                color = CLASS_COLORS.get(instance_class, DEFAULT_CLASS_COLOR)
+            else:
+                color_hsv = self._rgb_to_hsv(color)
+                # we use a minimum saturation of 100 to avoid too light colors,
+                # thus we increment saturation by step size of (256-100)/8.
+                saturation = (color_hsv[1] + 156 / 8) % 256
+                if saturation < 100:
+                    saturation += 100
+                color_hsv = (color_hsv[0], int(saturation), color_hsv[2])
+                color = self._hsv_to_rgb(color_hsv)
+            self.class_instance_color_state.update({instance_class: color})
+
+        return color
+
+    def _draw_contours_single_mask(
+        self,
+        masks: np.ndarray,
+        index: int,
+        image: np.ndarray,
+        instance_color: Tuple[int, int, int],
+    ) -> np.ndarray:
+        """Draws the contour around a single instance segmentation mask."""
+        ret_image = image.copy()
+        contour, _ = cv.findContours(masks[index], cv.RETR_TREE, cv.CHAIN_APPROX_SIMPLE)
+        # use a darker/lighter saturation of instance hue for the
+        # contour so that it is more visible
+        hsv = Node._rgb_to_hsv(instance_color)
+        contour_color_hsv = (hsv[0], int((hsv[1] + 127) % 256), hsv[2])
```

Comment:
Would fixing the colours of contours to black look better? Following Eric's screenshots from COCO website:

![image (6)](https://user-images.githubusercontent.com/33995028/176355505-920316cf-2646-46b3-81db-477b943797e6.png)

Reply:
Amended.